addons/viewport/src/components/Tool.tsx

```diff
@@ -0,0 +1,258 @@
+import { MINIMAL_VIEWPORTS } from '../defaults';
+import { ViewportAddonParameter, ViewportMap, ViewportStyles, Styles } from '../models';
+import { ViewportChangedEventProps } from '../models/ViewportAddonEvents';
+
+interface ViewportItem {
+  id: string;
+  title: string;
+  styles: Styles;
+  type: 'desktop' | 'mobile' | 'tablet' | 'other';
+  default?: boolean;
+}
+
+const toList = memoize(50)((items: ViewportMap): ViewportItem[] => [
+  ...baseViewports,
+  ...Object.entries(items).map(([id, { name, ...rest }]) => ({ ...rest, id, title: name })),
+]);
+
+const responsiveViewport: ViewportItem = {
+  id: 'reset',
+  title: 'Reset viewport',
+  styles: null,
+  type: 'other',
+};
+
+const baseViewports: ViewportItem[] = [responsiveViewport];
+
+const toLinks = memoize(50)((list: ViewportItem[], active: LinkBase, set, state, close): Link[] => {
+  return list
+    .map(i => {
+      switch (i.id) {
+        case responsiveViewport.id: {
+          if (active.id === i.id) {
+            return null;
+          }
+        }
+        default: {
+          return {
+            ...i,
+            onClick: () => {
+              set({ ...state, selected: i.id });
+              close();
+            },
+          };
+        }
+      }
+    })
+    .filter(Boolean);
+});
+
+const iframeId = 'storybook-preview-iframe';
+const wrapperId = 'storybook-preview-wrapper';
+
+interface LinkBase {
+  id: string;
+  title: string;
+  right?: ReactNode;
+  type: 'desktop' | 'mobile' | 'tablet' | 'other';
+  styles: ViewportStyles | ((s: ViewportStyles) => ViewportStyles) | null;
+}
+
+interface Link extends LinkBase {
+  onClick: () => void;
+}
+
+const flip = ({ width, height, ...styles }: ViewportStyles) => ({
+  ...styles,
+  height: width,
+  width: height,
+});
+
+const ActiveViewportSize = styled.div(() => ({
+  display: 'inline-flex',
+}));
+
+const ActiveViewportLabel = styled.div<{}>(({ theme }) => ({
+  display: 'inline-block',
+  textDecoration: 'none',
+  padding: 10,
+  fontWeight: theme.typography.weight.bold,
+  fontSize: theme.typography.size.s2 - 1,
+  lineHeight: '1',
+  height: 40,
+  border: 'none',
+  borderTop: '3px solid transparent',
+  borderBottom: '3px solid transparent',
+  background: 'transparent',
+}));
+
+const IconButtonWithLabel = styled(IconButton)(() => ({
+  display: 'inline-flex',
+  alignItems: 'center',
+}));
+
+const IconButtonLabel = styled.div<{}>(({ theme }) => ({
+  fontSize: theme.typography.size.s2 - 1,
+  marginLeft: 10,
+}));
+
+interface ViewportToolState {
+  isRotated: boolean;
+  selected: string | null;
+}
+
+const getStyles = (
+  prevStyles: ViewportStyles,
+  styles: Styles,
+  isRotated: boolean
+): ViewportStyles => {
+  if (styles === null) {
+    return null;
+  }
+  const result = typeof styles === 'function' ? styles(prevStyles) : styles;
+  return isRotated ? flip(result) : result;
+};
+
+type Props = {
+  theme: Theme;
+  api: API;
```

Comment:
I don't think there should be a need for this, if you can do `useStorybookApi` ?